ecosystem/typescript/sdk/src/transactions/transaction_worker.ts

```diff
@@ -0,0 +1,225 @@
+
+  // transactions that have been committed to chain
+  executedTransactions: Array<[string, bigint, any]> = [];
+
+  /**
+   * Provides a simple framework for receiving payloads to be processed.
+   *
+   * @param provider - a client provider
+   * @param sender - a sender as AptosAccount
+   * @param maxWaitTime - the max wait time to wait before resyncing the sequence number to the current on-chain state
+   * @param maximumInFlight - submit up to `maximumInFlight` transactions per account
+   * @param sleepTime - If `maximumInFlight` are in flight, wait `sleepTime` seconds before re-evaluating
+   */
+  constructor(
+    provider: Provider,
+    account: AptosAccount,
+    maxWaitTime: number,
+    maximumInFlight: number,
+    sleepTime: number,
+  ) {
+    super();
+    this.provider = provider;
+    this.account = account;
+    this.started = false;
+    this.stopped = false;
+    this.accountSequnceNumber = new AccountSequenceNumber(provider, account, maxWaitTime, maximumInFlight, sleepTime);
+  }
+
+  /**
+   * Gets the current account sequence number,
+   * generates the transaction with the account sequence number,
+   * adds the transaction to the outstanding transaction queue
+   * to be processed later.
+   */
+  async submitNextTransaction() {
+    if (this.transactionsQueue.length === 0) return;
+    const sequenceNumber = await this.accountSequnceNumber.nextSequenceNumber();
+    if (sequenceNumber === null) return;
+    const transaction = await this.generateNextTransaction(this.account, sequenceNumber);
+    if (!transaction) return;
+    const pendingTransaction = this.provider.submitSignedBCSTransaction(transaction);
+    this.outstandingTransactions.push([pendingTransaction, sequenceNumber]);
+  }
+
+  /**
+   * Reads the outstanding transaction queue and submits the transaction to chain.
+   *
+   * If the transaction has fulfilled, it pushes the transaction to the processed
+   * transactions queue and fires a transactionsFulfilled event.
+   *
+   * If the transaction has failed, it pushes the transaction to the processed
+   * transactions queue with the failure reason and fires a transactionsFailed event.
+   */
+  async processTransactions() {
+    const awaitingTransactions = [];
+    const awaitingSequenceNumbers = [];
+
+    while (this.outstandingTransactions.length > 0) {
+      const [pendingTransaction, sequenceNumber] = this.outstandingTransactions.shift()!;
+
+      awaitingTransactions.push(pendingTransaction);
+      awaitingSequenceNumbers.push(sequenceNumber);
+    }
+
+    // send awaiting transactions to chain
+    const sentTransactions = await Promise.allSettled(awaitingTransactions);
+
+    for (let i = 0; i < sentTransactions.length && i < awaitingSequenceNumbers.length; i += 1) {
+      // check sent transaction status
+      const sentTransaction = sentTransactions[i];
+      const sequenceNumber = awaitingSequenceNumbers[i];
+      if (sentTransaction.status === "fulfilled") {
```

Comment:
are there enums or consts that would be better here?

Reply:
not provided by Promise, but I can make it a const
```
interface PromiseFulfilledResult<T> {
    status: "fulfilled";
    value: T;
}

interface PromiseRejectedResult {
    status: "rejected";
    reason: any;
}

type PromiseSettledResult<T> = PromiseFulfilledResult<T> | PromiseRejectedResult;
```